src/Symfony/Component/HttpClient/Retry/HttpStatusCodeDecider.php

```diff
@@ -11,25 +11,56 @@
 
 namespace Symfony\Component\HttpClient\Retry;
 
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+
 /**
  * Decides to retry the request when HTTP status codes belong to the given list of codes.
  *
  * @author Jérémy Derussé <jeremy@derusse.com>
  */
 final class HttpStatusCodeDecider implements RetryDeciderInterface
 {
+    public const METHOD_DECISION = [
+        'GET' => true,
+        'HEAD' => true,
+        'PUT' => true,
+        'DELETE' => true,
+        'OPTIONS' => true,
+        'TRACE' => true,
+    ];
+    private const STATUS_CODES_DECISION = [
+        423 => true,
```

Comment:
DEFAULT_RETRY_CODES = [
    423, 425, 429, 500 => self::IDEMPOTENT_METHODS, etc.
];

Reply:
hmm.. I'm not sure mixing key and value is a good thing:
- need more Runtime cpu to check `isset` or `in_array` + `is_array`
- Is there a ways to provide a semantic configuration for this?

Reply:
> need more Runtime cpu to check isset or in_array + is_array

not the case for simple arrays like this

> Is there a ways to provide a semantic configuration for this?

that's decoupled, we could force a subset for the config

Reply:
done